code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+          walletLpProvider2.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+
+        // 4. Remove initial liquidity
+        const lpTokenId = hardCodedPool1.lpTokenId;
+        const lpTokenAccountInfo1 = await api.query.tokens.accounts(walletLpProvider1.publicKey, lpTokenId);
+        const availableLpTokens1 = lpTokenAccountInfo1.free;
+
+        const removeAssetsMinAmounts = api.createType("BTreeMap<u128, u128>", {
+          "4": Pica(0),
+          "130": USDT(0)
+        });
+
+        const expectedAmountLpTokens = await api.rpc.pablo.simulateAddLiquidity(
+          walletLpProvider1.address,
+          hardCodedPool1.poolId.toString(), // ToDo: Update pool id w/ created pool!
+          api.createType("BTreeMap<u128, u128>", removeAssetsMinAmounts)
+        );
+
+        const {
+          data: [resultWho, resultPoolId, resultAssetsAmount, resultMintedLp]
+        } = await sendAndWaitForSuccess(
+          api,
+          walletLpProvider1,
+          api.events.pablo.LiquidityRemoved.is,
+          // Parameters: poolId, assetsMap, minMintAmount, keepAlive
+          api.tx.pablo.removeLiquidity(hardCodedPool1.poolId, availableLpTokens1, removeAssetsMinAmounts) // ToDo: Update pool id w/ created pool!
+        );
+
+        // 5. Remove 2nd wallet liquidity
+        const lpTokenAccountInfo2 = await api.query.tokens.accounts(walletLpProvider2.publicKey, lpTokenId);
+        const availableLpTokens2 = lpTokenAccountInfo2.free;
+
+        await sendAndWaitForSuccess(
+          api,
+          walletLpProvider2,
+          api.events.pablo.LiquidityRemoved.is,
+          // Parameters: poolId, assetsMap, minMintAmount, keepAlive
+          api.tx.pablo.removeLiquidity(hardCodedPool1.poolId, availableLpTokens2, removeAssetsMinAmounts) // ToDo: Update pool id w/ created pool!
+        );
+
+        const finalQuoteAssetFundsLpProvider1 = await api.query.tokens.accounts(
+          walletLpProvider1.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+        const finalQuoteAssetFundsLpProvider2 = await api.query.tokens.accounts(
+          walletLpProvider2.publicKey,
+          hardCodedPool1.quoteAssetId
+        );
+
+        expect(
+          finalQuoteAssetFundsLpProvider1.free
+            .sub(quoteAssetFundsLpProvider1BeforeRemoval.free)
+            .add(finalQuoteAssetFundsLpProvider2.free.sub(quoteAssetFundsLpProvider2BeforeRemoval.free))
+            .sub(new BN((100_000_000_000).toString()))
+        ).to.be.bignumber.closeTo(
+          new BN(amountEarnedFeesQuoteAssetWallet1.toString()).add(new BN(priceFirstTrade.toFixed(0))),
+          new BN(1)
+        );
+      }
+    );
+
+    it(
+      "#3.8 After adding funds, and after a trade, transferring part of them to another wallet " +
+        "the initial wallet can withdraw the provided liquidity with received fees, " +
+        "while the second wallet won't receive fees",
```

Comment:
The same goes for this one as well. ı don't think we are separately holding a struct for fee distribution. I couldn't fully understand the assertions but I would say if we try to match the amounts received they would be the same for both wallets.

Reply:
True and the assertions are actually checking this. Need to fix test title & comment.
Thanks sir.